common/src/main/java/com/tencent/rss/common/ShuffleIndexDataItem.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Tencent is pleased to support the open source community by making
+ * Firestorm-Spark remote shuffle server available.
+ *
+ * Copyright (C) 2021 THL A29 Limited, a Tencent company.  All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at
+ *
+ * https://opensource.org/licenses/Apache-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package com.tencent.rss.common;
+
+import static com.tencent.rss.common.util.Constants.DEFAULT_PATH_PREFIX;
+
+/**
+ * ShuffleIndexDataItem is the index data read from server side in local mode or hdfs side in hdfs mode.
+ * It is guaranteed that only one index file at most in the server side in local mode, in which the
+ * dataPathPrefix filed is useless and set to default value. However there may be multiple index file in hdfs mode,
+ * and the dataPathPrefix the dataPathPrefix field should be a valid data path prefix can be used to get the
+ * data file indexed by this index item.
+ */
+public class ShuffleIndexDataItem {
```

Comment:
may ShuffleIndexItem be better? Because index and data is two concepts in our system, we shouldn't mix them.

Reply:
I feel that ShuffleIndexDataItem should be used in hdfs mode only.  In local mode, prefix path is weird.

Reply:
Agree to use ShuffleIndexItem instead. But path prefix is a common concept in both local and hdfs, this field can be empty in local is simple because we force only one data/index file to exist.

Reply:
But path inl local mode is  invisible to client.

Reply:
here is the problem, if only one data/index file is possible?
The current implementation is the work around if can't append data to hdfs, eg, lease problem
It may be time to improve it first